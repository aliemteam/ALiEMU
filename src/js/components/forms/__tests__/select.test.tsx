import React from 'react';
import renderer from 'react-test-renderer';

import Select from '../select';

// tslint:disable:react-a11y-role-has-required-aria-props

it('should render correctly', () => {
    const tree = renderer.create(
        <>
            <Select>
                <option>Foo</option>
                <option>Bar</option>
                <option>Baz</option>
            </Select>
            <Select label="Labelled Select">
                <option>Foo</option>
                <option>Bar</option>
                <option>Baz</option>
            </Select>
            <Select label="Disabled Label Select" disabled>
                <option>Foo</option>
                <option>Bar</option>
                <option>Baz</option>
            </Select>
        </>,
    );
    expect(tree).toMatchSnapshot();
});
