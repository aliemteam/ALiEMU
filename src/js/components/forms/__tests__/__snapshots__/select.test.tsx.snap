// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
Array [
  <select
    className=""
  >
    <option
      value=""
    />
    <option>
      Foo
    </option>
    <option>
      Bar
    </option>
    <option>
      Baz
    </option>
  </select>,
  <label
    className=""
  >
    Labelled Select
    <select
      className=""
    >
      <option
        value=""
      />
      <option>
        Foo
      </option>
      <option>
        Bar
      </option>
      <option>
        Baz
      </option>
    </select>
  </label>,
  <label
    className="undefined"
  >
    Disabled Label Select
    <select
      className=""
      disabled={true}
    >
      <option
        value=""
      />
      <option>
        Foo
      </option>
      <option>
        Bar
      </option>
      <option>
        Baz
      </option>
    </select>
  </label>,
]
`;
